sql/expression/arithmetic.go

```diff
@@ -330,14 +503,18 @@ func (a *Arithmetic) evalLeftRight(ctx *sql.Context, row sql.Row) (interface{},
 
 func (a *Arithmetic) convertLeftRight(left interface{}, right interface{}) (interface{}, interface{}, error) {
 	var err error
-	typ := a.Type()
+
+	typ := a.returnType(left, right)
 
 	if i, ok := left.(*TimeDelta); ok {
 		left = i
 	} else {
 		left, err = typ.Convert(left)
 		if err != nil {
-			return nil, nil, err
+			// TODO : any error here from converting should be added as warning
+			// the value is interpreted as 0, but we need to match the type of the other valid value
+			// to avoid additional conversion, the nil value is handled in each operation
+			left = nil
```

Comment:
Good comment. It should be pretty straightforward to set that warning. You'll just need to pass `ctx` into this function, then use the `ctx.Session.Warn()` function to set one. 

Would be awesome to have a test for this, too (e.g. `SELECT "abc"-1;`). I believe I added a field in ScriptTestAssertion that lets you test that the warning message is present and contains an expected substring.